src/pages/resultsView/cnSegments/CNSegments.tsx

```diff
@@ -1,139 +1,114 @@
     }
 
-    downloadSegmentData() {
-            const self = this;
-            const xhr = new XMLHttpRequest();
-            const sendData = `cancerStudyId=${self.props.studyId}&sampleIds=${self.props.sampleIds.join(',')}`;
-
-            xhr.onreadystatechange = function() {
-                if (xhr.readyState === 4 && xhr.status === 200) {
-                    const reader = new FileReader();
-
-                    reader.addEventListener('loadend', (e: any) => {
-                        // seems like the downloaded string is wrapped with quotes,
-                        // also it contains "\\n", "\\t", instead of "\n", and "\t"
-                        const text = unescapeTabDelimited(unquote(e.srcElement.result));
-                        fileDownload(text, self.props.studyId + '_segments.seg');
-                    });
-
-                    // this will fire a "loadend" event
-                    reader.readAsText(xhr.response);
-                }
-            };
-
-            // Post data to URL which handles post request
-            xhr.open("POST", buildCBioPortalPageUrl("api-legacy/segmentfile"));
-            xhr.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
-            xhr.responseType = 'blob';
-            xhr.send(sendData);
+    @computed get features() {
+        return this.props.store.cnSegments.result || [];
     }
-    
-    componentDidMount(){
-        const iframe: HTMLIFrameElement = document.createElement('iframe');
-        iframe.id = "cnSegmentsFrame";
-        $(iframe).css({
-           width:'100%',
-           height:630,
-           border:'none'
-        }).appendTo($(this.iframeDiv).empty());
-        if (iframe.contentWindow) {
-            iframe.contentWindow.document.open();
-            iframe.contentWindow.document.write(this.getIframeBody());
-            iframe.contentWindow.document.close();
-        }
-    }
-    
-    render(){
-        return (
-            <div>
-                <div style={{border:'1px solid #ddd', marginBottom:10 }} ref={(el:HTMLDivElement)=>this.iframeDiv = el}/>
-                <p>Download a copy number segment file for the selected samples
-                    &nbsp;<button onClick={this.downloadSegmentData} className="btn btn-default btn-sm">Download</button>
-                </p>
-            </div>
-        )
-        
-    }
-    
-}
-
-@observer
-export default class CNSegments extends React.Component<{ store: ResultsViewPageStore}, {}> {
 
-    @observable activeTabId:string;
+    @computed get filename()
+    {
+        let prefix = "";
+
+        if (this.props.store.studies.result) {
+            if (this.props.store.studies.result.length > 1) {
+                prefix = "multi_study_";
+            }
+            else if (this.props.store.studies.result.length === 1) {
+                prefix = `${this.props.store.studies.result[0].studyId}_`;
+            }
+        }
 
-    render(){
+        return `${prefix}segments.seg`;
```

Comment:
this one is fine I think because its kind of just a weak dependence. I don't think I understand what you're saying/how it fits into our larger discussion.